### PR TITLE
Batch update students

### PR DESCRIPTION
**Pull request by dting**
This is a bit rough around the edges but enables updating iep, cfa, and withdrawn properties of students in bulk. Currently only enabled on the dashboard view but could be refactored into some sort of grid menu service and enabled in other views with lists of students.

### CODE REVIEW

File: server/api/student/index.js [hunk 1]
```diff
@@ -24,6 +25,11 @@ router.put('/:studentId/iep', authorize, controller.updateIEP);
 router.put('/:studentId/cfa', authorize, controller.updateCFA);
 router.put('/:studentId/withdrawn', authorize, controller.updateWithdrawn);
 
+router.put('/:field',
+  authorizeBatch,
+  controller.validateBatchUpdate,
+  controller.batchUpdate);
```
Comment (pdotsani) [inline server/api/student/index.js]: So when you call two controllers, they run asynchronously?

File: server/api/student/student.controller.js [hunk 2]
```diff
@@ -73,6 +75,38 @@ exports.updateWithdrawn = function(req, res) {
...
+exports.batchUpdate = function(req, res) {
+  var updateValue = !!req.body.value;
+  var promises = _.map(req.students, function(student) {
+    student[req.field] = updateValue;
+    return student.save(function(err) {
+      if (err) return handleError(res, err);
+      return student;
+    });
+  });
+  Promise.all(promises).then(function(saved) {
```
Comment (pdotsani) [inline server/api/student/student.controller.js]: Are these built in promises?

File: client/app/main/dashboard/dashboard.controller.js [hunk 3]
```diff
@@ -37,7 +126,8 @@ function DashboardCtrl($scope, $timeout, AbsenceRecord, GridDefaults, Student) {
         $scope.loading = false;
         $timeout($scope.gridApi.treeBase.expandAllRows);
       });
-      $scope.menuItems.length = 1;
+      $scope.menuItems.length = 0;
+      [].push.apply($scope.menuItems, defaultMenuItems);
```
Comment (pdotsani) [inline client/app/main/dashboard/dashboard.controller.js]: I get the basics of commands like `call` and `apply`, any other resources/tutorials that you can recommend on this subject?

**Comment by pdotsani** [comment]
lgtm. Just a couple of questions.

File: server/api/student/index.js (hunk 1 above)
Comment (dting) [inline server/api/student/index.js]: no, they are run in order.

File: server/api/student/student.controller.js (hunk 2 above)
Comment (dting) [inline server/api/student/student.controller.js]: Document#save's return a mongoose promise.

File: client/app/main/dashboard/dashboard.controller.js (hunk 3 above)
Comment (dting) [inline client/app/main/dashboard/dashboard.controller.js]: MDN explains it fairly well.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call

**Pull request merged**